### PR TITLE
EE - midi output choice. (credits kelv)

### PR DESCRIPTION
**Pull request by Langerz82**
Credits: Kelv

Select MIDI Output choice for Retroarch.

depends on:
https://github.com/EmuELEC/emuelec-emulationstation/pull/97

### CODE REVIEW

**Comment by Langerz82** [comment]
I could make this PR, so that it's start the midi process only once and keeps it in memory, rather than killing/restarting the process every game launched what does everyone think?
Only the fluidsynth might need this option since it needs a delay to get the config value.

**Comment by Langerz82** [comment]
tested briefly.

**Comment by Kelvfimer** [comment]
Thx @Langerz82 i can test it by next week

**Comment by Langerz82** [comment]
for kelv to test with dosbox

**Comment by Langerz82** [comment]
apologies for the several changes, I made some minor code design changes.

File: packages/sx05re/emuelec/bin/emuelecRunEmu.sh [hunk 1]
```diff
@@ -110,6 +110,11 @@ if [[ "${EMULATOR}" = "retrorun" ]]; then
     LIBRETRO=""
 fi
 
+MIDI_OUTPUT="$(get_ee_setting "ra_midi_output" "${PLATFORM}" "${ROMNAME}")"
```
Comment (Kelvfimer) [inline packages/sx05re/emuelec/bin/emuelecRunEmu.sh:113]: it must be MIDI_OUTPUT=$(get_ee_setting "ra_midi_output" "${PLATFORM}" "${ROMNAME##*/}")  removing the quotation marks and using that ROMNAME parameter @Langerz82

File: packages/sx05re/emuelec/bin/emuelec-utils [hunk 2]
```diff
@@ -6,6 +6,47 @@
...
+
+  [[ "${midi_emulator}" == "retrorun" ]] && return
+
+  local midi_output=OFF
+  
+  case ${midi_source} in
+    fluidsynth)
+      fluidsynth -is /storage/roms/bios/sc/SC-55.sf2 &
+      sleep 3
+      =$(cat /proc/asound/seq/clients | grep "Port   0 : \"S" | cut -d'"' -f2)
```
Comment (Kelvfimer) [inline packages/sx05re/emuelec/bin/emuelec-utils:32]: it must be midi_output=$(cat /proc/asound/seq/clients | grep "Port   0 : "S" | cut -d'"' -f2) it lacks the midi_ouput var

File: packages/sx05re/emuelec/bin/emuelec-utils [hunk 3]
```diff
@@ -6,6 +6,47 @@
...
+      [[ -z "${MIDI_OUTPUT}" ]] && midi_output=OFF
```
Comment (Kelvfimer) [inline packages/sx05re/emuelec/bin/emuelec-utils:33]: You should remove this line @Langerz82

File: packages/sx05re/emuelec/bin/emuelec-utils [hunk 4]
```diff
@@ -6,6 +6,47 @@
...
+      ;;
+    timidity)
+      timidity -iA --module 1 &
+      midi_output="TiMidity port 0"
+			;;
+    mt32d)
+      mt32d -o 1 -i 24 -f /roms/bios &
```
Comment (Kelvfimer) [inline packages/sx05re/emuelec/bin/emuelec-utils:40]: Here I was using -o 1 to use CM32-L roms that it is the mt32d latest device that was released. I think we should add and extra entry in emulationstation to select between mt32d original version and CM32-L as there are games that were programmed to use the original one. Original version flag is -o 2 . Let's discuss on discord @Langerz82  :)

File: packages/sx05re/emuelec/bin/emuelecRunEmu.sh [hunk 5]
```diff
@@ -498,6 +503,9 @@ fi
 
 emuelec-utils end_app_video
 
+# Kill MIDI Processes
+emuelec-utils killall_midi
```
Comment (Kelvfimer) [inline packages/sx05re/emuelec/bin/emuelecRunEmu.sh:507]: it is needed to set back midi_output = "OFF" in retroarch.cfg . I see that current behaviout is keeping the lastedt midi_output used and that's not ok. The midi_ouput must come back to midi_output = "OFF" @Langerz82

**Pull request merged**